MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSourceProtocol.swift

```diff
@@ -2,13 +2,13 @@ import Combine
 
 protocol MovieNetworkDataSourceProtocol {
 
-    var popularMovies: AnyPublisher<[MovieDataModel], Never> { get }
+    var popularMovies: AnyPublisher<[MovieDataModel], RequestError> { get }
```

Comment:
Is this `RequestError` specific to `client` layer?

Reply:
Created MovieDataError that is specific to DS layer so I don't use RequestError in DS layer anymore.